homeassistant/components/vicare/fan.py

```diff
@@ -0,0 +1,192 @@
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ViCare fan platform."""
+
+    device_list = hass.data[DOMAIN][config_entry.entry_id][DEVICE_LIST]
+
+    async_add_entities(
+        await hass.async_add_executor_job(
+            _build_entities,
+            device_list,
+        )
+    )
+
+
+class ViCareFan(ViCareEntity, FanEntity):
+    """Representation of the ViCare ventilation device."""
+
+    _attr_speed_count = len(ORDERED_NAMED_FAN_SPEEDS)
+    _attr_supported_features = FanEntityFeature.SET_SPEED | FanEntityFeature.PRESET_MODE
+    _attributes: dict[str, Any] = {
+        "active_vicare_mode": None,
+        "active_vicare_program": None,
+        "vicare_modes": None,
+        "vicare_programs": None,
+    }
+
+    def __init__(
+        self,
+        device_config: PyViCareDeviceConfig,
+        device: PyViCareDevice,
+        translation_key: str,
+    ) -> None:
+        """Initialize the fan entity."""
+        super().__init__(device_config, device, translation_key)
+        self._attr_translation_key = translation_key
+
+        with suppress(PyViCareNotSupportedFeatureError):
+            self._attributes["vicare_modes"] = self._api.getAvailableModes()
+        self._attributes["vicare_programs"] = self._api.getAvailablePrograms()
+
+    def update(self) -> None:
+        """Update state of fan."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attributes["active_vicare_mode"] = self._api.getActiveMode()
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attributes["active_vicare_program"] = self._api.getActiveProgram()
+        except RequestConnectionError:
+            _LOGGER.error("Unable to retrieve data from ViCare server")
+        except ValueError:
+            _LOGGER.error("Unable to decode data from ViCare server")
+        except PyViCareRateLimitError as limit_exception:
+            _LOGGER.error("Vicare API rate limit exceeded: %s", limit_exception)
+        except PyViCareInvalidDataError as invalid_data_exception:
+            _LOGGER.error("Invalid data from Vicare server: %s", invalid_data_exception)
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        raise ServiceValidationError(
+            translation_domain=DOMAIN,
+            translation_key="fan_must_be_on",
+        )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the entity is on."""
+        # Viessmann ventilation unit cannot be turned off
+        return True
+
+    @property
+    def percentage(self) -> int | None:
+        """Return the current speed percentage."""
+
+        # if fan mode is something else than *permanent*, speed is controlled by sensor/schedule
+        if self._attributes["active_vicare_mode"] is not VentilationMode.PERMANENT:
```

Comment:
You can not compare with is not here. You have to use != 
The same applies to the other comparisons

Reply:
We can also show the active percentage in the Modes VENTILATION and SENSOR_OVERRIDE. The app shows also the four different values in this modes.

Reply:
done

Reply:
Perfect. Can you change the comparison to `        if self._attributes["active_vicare_mode"] != str(VentilationMode.PERMANENT):
` because we have two different types of objects here. So comparing with is Not is always false. After your commit I´ll give it another try.